### PR TITLE
Add release to CircleCI

### PR DESCRIPTION
**Pull request by SuperQ**
* Add release process to CircleCI.
* Remove Travis configuration.

https://github.com/cloudalchemy/auto-maintenance/issues/2

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,18 +41,91 @@ jobs:
...
+           ferrarimarco/github-changelog-generator:1.15.2 \
+           --user "${CIRCLE_PROJECT_USERNAME}" \
+           --project "${CIRCLE_PROJECT_REPONAME}" \
+           --token "${GH_TOKEN}" \
+           --release-url "https://galaxy.ansible.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME#ansible-}" \
+           --unreleased-label "**Next release**" --no-compare-link \
+           --future-release "${GIT_TAG}"
+      - run: git add CHANGELOG.md
+      - run: git commit -m "[ci skip] Automatic changelog update"
+      - run: git tag "${GIT_TAG}" -a -m "Automatic tag generation for CI build no. ${CIRCLE_BUILD_NUM}"
```
Comment (paulfantom) [inline .circleci/config.yml:79]: Do we need this when using `ghr`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,18 +41,91 @@ jobs:
...
+      - run: git push "https://${GH_TOKEN}:@${GIT_URL}" --tags || circleci-agent step halt
```
Comment (paulfantom) [inline .circleci/config.yml:80]: Do we need this when using `ghr`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -32,18 +41,91 @@ jobs:
...
+      - run: git push "https://${GH_TOKEN}:@${GIT_URL}" || circleci-agent step halt
```
Comment (paulfantom) [inline .circleci/config.yml:81]: Probably should be before pushing tags

File: .circleci/config.yml [hunk 4]
```diff
@@ -32,18 +41,91 @@ jobs:
...
+      - run: |
+          ghr \
+            -t ${GH_TOKEN} \
+            -u ${CIRCLE_PROJECT_USERNAME} \
+            -r ${CIRCLE_PROJECT_REPONAME} \
+            -n ${GIT_TAG} \
+            -b "$(sed -n -e '/## \[0.22.0\]/,/## \[/ p' CHANGELOG.md | sed -e '$ d')" \
+            ${GIT_TAG} ./artifacts/  # We don't have artifacts but ghr still needs sth
```
Comment (paulfantom) [inline .circleci/config.yml:89]: I am not sure if this needs an artifacts path or if this is optional

**Comment by paulfantom** [comment]
Possibly next step would be to optimize workflow with preserving python cache.

File: .circleci/config.yml (hunk 2 above)
Comment (SuperQ) [reply .circleci/config.yml:80]: Good question, I don't know.

**Review by SuperQ — COMMENTED** [review]
LGTM

**Pull request merged**